utils/steward-cli/src/commands/actions/add_to_blacklist.rs

```diff
@@ -1,35 +1,73 @@
 use std::sync::Arc;
 
-use anchor_lang::{InstructionData, ToAccountMetas};
+use anchor_lang::{AccountDeserialize, InstructionData, ToAccountMetas};
 use anyhow::Result;
 use solana_client::nonblocking::rpc_client::RpcClient;
 use solana_program::instruction::Instruction;
 
+use crate::utils::transactions::{configure_instruction, maybe_print_tx};
 use solana_sdk::{
     pubkey::Pubkey, signature::read_keypair_file, signer::Signer, transaction::Transaction,
 };
-use stakenet_sdk::utils::transactions::{configure_instruction, print_base58_tx};
 
 use crate::commands::command_args::AddToBlacklist;
+use stakenet_sdk::utils::accounts::get_validator_history_address;
+use validator_history::{self, ValidatorHistory};
 
 pub async fn command_add_to_blacklist(
     args: AddToBlacklist,
     client: &Arc<RpcClient>,
     program_id: Pubkey,
 ) -> Result<()> {
-    // Creates config account
     let authority = read_keypair_file(args.permissioned_parameters.authority_keypair_path)
         .expect("Failed reading keypair file ( Authority )");
 
+    let authority_pubkey = if args.permissioned_parameters.transaction_parameters.print_tx
+        || args
+            .permissioned_parameters
+            .transaction_parameters
+            .print_gov_tx
+    {
+        let config_account = client
+            .get_account(&args.permissioned_parameters.steward_config)
+            .await?;
+        let config = jito_steward::Config::try_deserialize(&mut config_account.data.as_slice())?;
+        config.blacklist_authority
+    } else {
+        authority.pubkey()
+    };
+
+    // Build list of indices, starting with those passed directly
+    let mut indices = args.validator_history_indices_to_blacklist.clone();
+    // Fetch indices for each vote account provided
+    println!("Vote Account\tHistory Address\tIndex");
+    for vote_account in args.vote_accounts_to_blacklist.iter() {
+        let history_address = get_validator_history_address(vote_account, &validator_history::id());
+        let (vh_index, account_exists) = match client.get_account(&history_address).await {
+            Ok(account) => match ValidatorHistory::try_deserialize(&mut account.data.as_slice()) {
+                Ok(vh) => (vh.index.to_string(), true),
+                Err(_) => ("N/A".to_string(), false),
+            },
+            Err(_) => ("N/A".to_string(), false),
+        };
+        println!(
+            "{}\thttps://solscan.io/account/{}\t{}",
+            vote_account, history_address, vh_index
+        );
+        if account_exists {
+            indices.push(vh_index.parse().unwrap());
```

Comment:
```suggestion
            indices.push(vh_index.parse()?);
```
I think we can avoid unwrapping here.